pkg/repo/index.go

```diff
@@ -251,7 +251,7 @@ func IndexDirectory(dir, baseURL string) (*IndexFile, error) {
 	for _, arch := range archives {
 		fname, err := filepath.Rel(dir, arch)
 		if err != nil {
-			return index, err
+			return nil, err
```

Comment:
This was intended as we want to return the current state of the index during the parsing stage. If it fails halfway through, the user may want to know what was indexed prior to failure.